oracle/oracle_test.go

```diff
@@ -93,6 +96,133 @@ func Test_Oracle_ManualSubscription(t *testing.T) {
+		PoolAddress:           "0xdead000000000000000000000000000000000000",
+		UpdaterAddress:        "",
+		DeployedSlot:          uint64(50000),
+		CheckPointSizeInSlots: uint64(100),
+		PoolFeesPercent:       5,
+		PoolFeesAddress:       "0xfee0000000000000000000000000000000000000",
+		CollateralInWei:       big.NewInt(1000000),
+	})
+
+	subsIndex := make([]uint64, 0)
+	totalAssets := big.NewInt(0)
+	const seed int64 = 50000
+	rand.Seed(seed)
+	fmt.Printf("Execution SEED: %d\n", seed)
+	// main loop, iterates through 100 slots
+	for i := 0; i <= 99; i++ {
+		newSubscription := make([]Subscription, 0)
+		newUnsubscription := make([]Unsubscription, 0)
+		don := make([]Donation, 0)
+		fmt.Println("NEW BLOCK:")
+
+		//throw dice to determine if a new subscription is set in this slot. 1/2 chance
+		dice := rand.Intn(2)
+		if dice == 0 {
+			newSubscription = GenerateSubsctiptions(
+				/*valIndexs*/ []uint64{50000 + uint64(i)},
+				/*valKeys*/ []string{"val" + strconv.FormatUint(50000+uint64(i), 10)},
+				/*collaterals*/ []*big.Int{big.NewInt(1000000)},
+				/*blockNums*/ []uint64{50000 + uint64(i)},
+				/*txHashes*/ []string{"0x1"},
+				/*depositAddrs*/ []string{"0xaaa0000000000000000000000000000000000000"},
+			)
+			subsIndex = append(subsIndex, newSubscription[0].ValidatorIndex)
+			totalAssets.Add(totalAssets, newSubscription[0].Collateral)
+		}
+
+		//throw dice to determine if a new unsubscription is set in this slot. 1/3 chance
+		//(can only unsubscribe already subbed validators)
+		dice = rand.Intn(3)
+		if dice == 0 && len(subsIndex) > 0 {
+			indexRandom := rand.Intn(len(subsIndex))
+			valtoUnsub := subsIndex[indexRandom]
+			newUnsubscription = GenerateUnsunscriptions(
+				/*valIndexs*/ []uint64{valtoUnsub},
+				/*valKeys*/ []string{"val" + strconv.FormatUint(valtoUnsub, 10)},
+				/*sender*/ []string{strconv.FormatUint(50000+uint64(i), 10)},
+				/*blockNums*/ []uint64{50000 + uint64(i)},
+				/*txHashes*/ []string{"0x1"},
+				/*depositAddrs*/ []string{strconv.FormatUint(50000+uint64(i), 10)},
+			)
+			//unsubsIndex = append(unsubsIndex, newUnsubscription[0].ValidatorIndex)
+
+			//delete subbed validator from slice that keeps all subbed validators
+			subsIndex = append(subsIndex[:indexRandom], subsIndex[indexRandom+1:]...)
+		}
+		//throw dice to determine if a new donation is set in this slot. 1/5 chance
+		dice = rand.Intn(5)
+		if dice == 0 {
+			donationAmount := big.NewInt(int64(rand.Intn(1000) + 10000))
+			newDonation := Donation{
+				AmountWei: donationAmount,
+				Block:     50000,
+				TxHash:    "my_tx_hash",
+			}
+			don = append(don, newDonation)
+			totalAssets.Add(totalAssets, donationAmount)
+		}
+
+		//throw dice to determine block type (ok, missed, wrongfee)
+		dice = rand.Intn(3)
+		//valToPropose := subsIndex[rand.Intn(len(subsIndex))]
+
+		//choose randomly a validator to propopse the block (can be an unsubbed validator, so we check automatic subs)
+		valToPropose := uint64(rand.Intn(101) + 50000)
```

Comment:
i assume the 101 is hardcoded because we are iterating 100 slots? can be have it configurable. so that i can run this for n slots (see the 99 in the for loop)

Reply:
Yes. I changed it so number of slots to simulate is now configurable by tester and passed through flags. In golang test, flag parsing is a little different: https://blog.jbowen.dev/2019/08/using-go-flags-in-tests/ (basically, I understood that flags must be global in tests.

New flag usage example: _$ go test -v -run Test_100_slots_test -iterations=10_

Reply:
mind removing this flag from the test? seems a bit weird, never seen this. so lets have it configurable but with just a single hardcoded variable at the begining of the test `numBlocks := 100`. i mean, what i meant is that there should be a single place to configure the amount of blocks (not multiple like before, where we had it in the for + rand.Intn(xx) etc.